verifier/src/lib.rs

```diff
@@ -17,8 +17,16 @@ pub use winterfell::StarkProof;
 /// Returns Ok(()) if the specified program was executed correctly against the specified inputs
 /// and outputs.
 ///
-/// Specifically, verifies that if a program with the specified `program_hash` is executed with the
-/// provided `public_inputs` and some secret inputs, and the result is equal to the `outputs`.
+/// Specifically, verifies that if a program with the specified `program_hash` is executed against
+/// the provided `stack_inputs` and some secret inputs, the result is equal to the `stack_outputs`.
+///
+/// Stack inputs are expected to be provided in the order as if they were pushed onto the stack
+/// one by one. Thus, the last value in the `stack_inputs` slice is expected to be the value at
+/// the top of the stack.
+///
+/// Stack outputs are expected to be provided in the order as if they were popped off the stack
+/// one by one. Thus, the value at the top of the stack is expected to be be in the first position
+/// of the `stack_outputs` slice. This the reverse of order of `stack_inputs` slice.
```

Comment:
I think it might be helpful to also say explicitly that the provided stack outputs are expected to match the order of the elements on the stack so that people don't think this is more complicated than it is. Maybe something like:

"...Thus, the value at the top of the stack is expected to be be in the first position of the `stack_outputs` slice, and the order of the rest of the output elements will also match the order on the stack. This the reverse..."

Reply:
Updated!